R/provider-gemini.R

```diff
@@ -65,7 +77,16 @@ method(chat_request, ProviderGemini) <- function(provider,
 
 
   req <- request(provider@base_url)
-  req <- req_headers_redacted(req, "x-goog-api-key" = provider@api_key)
+  if (nchar(provider@api_key)) {
+    req <- req_headers_redacted(req, "x-goog-api-key" = provider@api_key)
+  } else {
+    # TODO: Can use req_headers_redacted() when !!! is supported.
```

Comment:
Actually once we bump to the next version of httr2, `req_headers()` will auto-redact `Authorization`.